app/helpers/application_helper.rb

```diff
@@ -187,6 +187,14 @@ def courses
     ]
   end
 
+  def custom_button
+    if current_user
+      render 'static_pages/home/curriculum_button'
+    else
+      render 'static_pages/home/sign_up_button'
```

Comment:
Creating partials for individual buttons is overkill imo. You should be able to access the `curriculum_button` and `sign_up_button` helpers here